### DISCUSSION TITLE
Update useSWR and useSWRInfinite to return same response references if nothing has changed

### DISCUSSION THREAD

**Discussion by rikutiira** [vercel/swr#2793] [Ideas]
Currently if you call the same `useSWR` hook twice (e.g. when component re-renders), the returned response is not the same reference as what the previous call returned even though nothing has changed. Using the returned response with hooks such as `useEffect`, `useMemo` or `useCallback` will therefore cause unnecessary computations to happen.

I would like to open a PR to update `useSWR` and `useSWRInfinite` to return the same response reference if nothing has changed, which is as simple as wrapping the returned response object in `useMemo` call.

I temporarily patched SWR for our project but hopefully this could be merged to SWR soon as it's a very quick and small fix.

**Comment by rikutiira** [comment]
Hey, is it okay to open a PR about this? 😅

**Comment by stem-ui** [comment]
@rikutiira can you share what you did to patch this?
